src/utilities/configToJSONSchema.ts

```diff
@@ -432,27 +428,42 @@ export function entityToJSONSchema(config: SanitizedConfig, incomingEntity: Sani
     });
   }
 
+  // used for relationship fields, to determine whether to use a string or number type for the ID
+  const collectionIDFieldTypes: { [key: string]: 'string' | 'number' } = config.collections.reduce((acc, collection) => {
```

Comment:
This came from [this removed function](https://github.com/payloadcms/payload/pull/2827/files#diff-c367a8ce2430ab72fd6a3b7af83f561885e3833586fc62ec402c0d368283f41bL25). Now we return an object for all collection id field types in the shape of:
```ts
{
  pages: 'string' | 'number',
  posts: 'string' | 'number',
}
```